### ISSUE TITLE
Don't disable .gitignore'd files defined beyond the first parent directory that contains a git repository

### ISSUE THREAD

**Issue by nfrasser** [zed-industries/community#575]
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

I set up my home directory as a git repository to sync my shell config between machines. I have a .gitignore there which includes my code projects (separate git repos). After setting this up, I noticed that any repo I open Zed has every file greyed out in the sidebar as if it was .gitignore'd. 

Steps to reproduce:

1. Create a git repository in the home directory with a .gitignore file
   ```sh
   cd ~
   git init
   echo "*.txt" > .gitignore
   ```
2. Create a git repo with a single file in the home directory
   ```sh
   mkdir example-repo
   cd example-repo
   git init
   touch example.txt
   ```
3. Open the repo in Zed
   ```
   zed .
   ```

I suspect this was introduced with the fix for #85 

### Expected behavior

I expect `example.txt` file to be available in the sidebar. Instead it is greyed out even though it is not included in the project's .gitignore (but is in the parent directory's). 

This also means I can't use `Cmd+P` to open file patterns ignored in the parent git repository but not the project git repository.

### Environment

Zed 0.59.0 – /Applications/Zed.app 
macOS 12.6 
architecture x86_64

**Comment by vaidotasp** [comment]
heya, it would be awesome to get this issue in the pipeline at some point 🙏 Likely not a priority for most but for some of us it is a bummer.

**Comment by nfrasser** [comment]
Figured out a somewhat hacky workaround that may stop working if #1378 is fixed:

My home directory is set up like this:

```
~
├── ...
├── .git
├── .gitconfig
├── .gitignore
├── .gitignore_global
├── Development
└── ...
```

All code projects are in the `Development` folder, which was previous ignored in `.gitignore`. Here's what I did:

1. Remove `Development` from `.gitignore`
3. Create a `.gitignore_global` file and add the line `/Development`
3. Set `.gitignore_global` to be the global ignores file:
   ```sh
   git config --global core.excludesfile ~/.gitignore_global
   ```

**Comment by younux** [comment]
Hello, 

I have the same problem. I have a bare repo that I clone to my HOME directory

```
git clone --bare https://github.com/younux/dotfiles.git "$HOME/.dotfiles"
git --git-dir="$HOME/.dotfiles" --work-tree="$HOME" checkout
git --git-dir="$HOME/.dotfiles" --work-tree="$HOME" submodule update --init --recursive
```

With a .gitignore containing `/**` at my HOME directory, I have two features not working at all:
- The git status was not showing in the project panel for all my projects (they are located in a sub directory of HOME)
- The search in projects not working at all (projects located in a sub directory of HOME))

**Comment by hferreiro** [comment]
@younux: as a workaround, you can use `.git/info/exclude` instead. It seems `zed` is currently not reading that file.

**Comment by younux** [comment]
thanks @hferreiro 
What I did is I removed the gitignore from my dotfiles repo and I set git config status.showUntrackedFiles to no locally

**Comment by osiewicz** [comment]
Hey, that's gonna be fixed with tomorrow's (18.10) Preview. :)
